fast_llm/data/dataset/gpt/config.py

```diff
@@ -160,6 +165,41 @@ class GPTBlendedDatasetConfig(BlendedDatasetConfig, GPTSampledDatasetConfig):
     datasets: list[GPTSampledDatasetConfig] = FieldUpdate()
 
 
+@config_class()
+class GPTComposedDatasetConfig(GPTIndexedDatasetConfig):
```

Comment:
```suggestion
class GPTPathBasedDatasetConfig(GPTIndexedDatasetConfig):
```
mainly because this name emphasizes the dependency of the dataset configuration on a path. The purpose is immediately clear to developers and users alike: This is for datasets defined by the contents of a directory. "Composed" is too generic. All datasets are composite in one way or another.

Reply:
Agreed this can be changed, but I'd prefer something a bit more descriptive still, to distinguish it from other path-based options we already have (memmap, legacy, json) and may have in the future. So anything that sounds enough like "Automated concatenation of memmap datasets in a directory based in an index file". Also I'd focus more on the `type_`, the class name is less important. 

Some other names I had considered previously considered but didn't choose for conciseness:
* `concatenated_memmap` 
* `auto_concatenate`
* `multi_memmap`

What do you think?

Reply:
`concatenated_memmap` then